pkgs/cli/src/main.zig

```diff
@@ -277,5 +307,206 @@ pub fn main() !void {
+    try ENR.decodeTxtInto(&node_enr, bootnodes[self_node_index]);
+
+    // Overriding the IP to 0.0.0.0 to listen on all interfaces
+    try node_enr.kvs.put("ip", "\x00\x00\x00\x00");
+
+    var node_multiaddrs = try node_enr.multiaddrP2PQUIC(allocator);
+    defer node_multiaddrs.deinit(allocator);
+    const listen_addresses = try node_multiaddrs.toOwnedSlice(allocator);
+    // these addresses are converted to a slice in the `run` function of `EthLibp2p` so it can be freed safely after `run` returns
+    defer {
+        for (listen_addresses) |addr| addr.deinit();
+        allocator.free(listen_addresses);
+    }
+
+    var connect_peer_list: std.ArrayListUnmanaged(Multiaddr) = .empty;
+    defer connect_peer_list.deinit(allocator);
+
+    for (bootnodes, 0..) |n, i| {
+        if (i != self_node_index) {
+            var n_enr: ENR = undefined;
+            try ENR.decodeTxtInto(&n_enr, n);
+            var peer_multiaddr_list = try n_enr.multiaddrP2PQUIC(allocator);
+            defer peer_multiaddr_list.deinit(allocator);
+            const peer_multiaddrs = try peer_multiaddr_list.toOwnedSlice(allocator);
+            defer allocator.free(peer_multiaddrs);
+            try connect_peer_list.appendSlice(allocator, peer_multiaddrs);
+        }
+    }
+
+    const connect_peers = try connect_peer_list.toOwnedSlice(allocator);
+    defer {
+        for (connect_peers) |addr| addr.deinit();
+        allocator.free(connect_peers);
+    }
+
+    var logger = utilsLib.getScopedLogger(.default, options.console_log_level, utilsLib.FileBehaviourParams{ .fileActiveLevel = options.log_file_active_level, .filePath = options.log_filepath, .fileName = options.log_filename });
+
+    network.* = try networks.EthLibp2p.init(allocator, loop, .{ .networkId = 0, .listen_addresses = listen_addresses, .connect_peers = connect_peers }, &logger);
+    try network.run();
+    const backend = network.getNetworkInterface();
+
+    var clock = try allocator.create(Clock);
+    clock.* = try Clock.init(allocator, chain_config.genesis.genesis_time, loop);
+
+    var beam_node = try BeamNode.init(allocator, .{
+        // options
+        .nodeId = node_id,
+        .config = chain_config,
+        .anchorState = anchorState,
+        .backend = backend,
+        .clock = clock,
+        .db = .{},
+        .validator_ids = validator_indices,
+        .logger = &logger,
+    });
+
+    try beam_node.run();
+    std.debug.print("Lean node {d} listened on {?d}\n", .{ node_id, try node_enr.getQUIC() });
+    try clock.run();
+}
+
+fn genesisConfigFromYAML(config: yaml.Yaml) !types.GenesisSpec {
+    const genesis_spec: types.GenesisSpec = .{
+        .genesis_time = @intCast(config.docs.items[0].map.get("GENESIS_TIME").?.int),
+        .num_validators = @intCast(config.docs.items[0].map.get("VALIDATOR_COUNT").?.int),
+    };
+    return genesis_spec;
+}
+
+fn nodesFromYAML(allocator: std.mem.Allocator, nodes_config: yaml.Yaml) ![]const []const u8 {
```

Comment:
create a util.zig file in cli/src if not already there and move these nodes/validators parsing fns over there

Reply:
moved to `node.zig`